hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -430,7 +440,7 @@ protected boolean needsBalance(Cluster cluster) {
     System.arraycopy(tempFunctionCosts, 0, curFunctionCosts, 0, this.curFunctionCosts.length);
 
     double initCost = currentCost;
-    double newCost = currentCost;
+    double newCost;
```

Comment:
What happened here? Seems like an extraneous change and maybe wrong?

Reply:
Oh this is anyways a redundant assignment, not an issue. It's not being used anywhere other than for loop and inside that loop, it is always assigned with new value. In fact, this was removed from trunk and branch-2 as well. I am confident this is not problematic at all.